controllers/cbcontainersagent_controller.go

```diff
@@ -104,8 +121,31 @@ func (r *CBContainersAgentController) Reconcile(ctx context.Context, req ctrl.Re
 		return ctrl.SetControllerReference(cbContainersAgent, controlledResource, r.Scheme)
 	}
 
+	accessToken, err := r.getAccessToken(context.Background(), cbContainersAgent)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	operatorVersion, err := r.OperatorVersionProvider.GetOperatorVersion()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	m, err := r.apiGateway(cbContainersAgent, accessToken).GetCompatibilityMatrixEntryFor(operatorVersion)
```

Comment:
@asankov-cb Maybe we should move this logic to the processor?

Reply:
not sure what you mean. check the latest code to see whether we are on the same page.

Reply:
@asankov-cb 

The calls to the api gateway are on the controller, why not do that from the processor, making:
- The controller responsible only on watching for CR changes (similar to controllers on the backend listening to http requests)
- The processor responsible on business logic like what you added (this is also nice because we already have there a api gateway interface)